.buildkite/scripts/steps/esql_grammar_sync.sh

```diff
@@ -0,0 +1,131 @@
+  destination_file="./packages/kbn-monaco/src/esql/antlr/esql_parser.g4"
+
+  # Copy the file
+  cp "$source_file" "$destination_file" 
+
+  # Insert the license header
+  temp_file=$(mktemp)
+  printf "%s\n\n// DO NOT MODIFY THIS FILE BY HAND. IT IS MANAGED BY A CI JOB.\n\n%s" "$license_header" "$(cat ${destination_file})" > "$temp_file"
+  mv "$temp_file" "$destination_file"
+
+  # Replace the line containing "parser grammar" with "parser grammar esql_parser;"
+  sed -i -e 's/parser grammar.*$/parser grammar esql_parser;/' "$destination_file" 
+
+  # Replace options {tokenVocab=EsqlBaseLexer;} with options {tokenVocab=esql_lexer;}
+  sed -i -e 's/options {tokenVocab=EsqlBaseLexer;}/options {tokenVocab=esql_lexer;}/' "$destination_file" 
+
+  echo "File copied and modified successfully."
+}
+
+report_main_step () {
+  echo ""
+  echo "-------------------------------------------------"
+  echo "MAIN STEP: $1"
+  echo "-------------------------------------------------"
+}
+
+main () {
+  cd "$PARENT_DIR" 
+
+  report_main_step "Cloning repositories"
+
+  rm -rf elasticsearch
+  git clone https://github.com/elastic/elasticsearch --depth 1 
+
+  rm -rf open-source
+  git clone https://github.com/elastic/open-source --depth 1 
+
+  cd "$KIBANA_DIR" 
+
+  license_header=$(cat "$PARENT_DIR/open-source/legal/elastic-license-2.0-header.txt")
+
+  report_main_step "Synchronizing lexer grammar..."
+  synchronize_lexer_grammar "$license_header"
+
+  report_main_step "Synchronizing parser grammar..."
+  synchronize_parser_grammar "$license_header"
+
+  # Check for differences
+  set +e
+  git diff --exit-code --quiet "$destination_file"
+  set -e
+
+  if [ $? -eq 0 ]; then
+    echo "No differences found. Our work is done here."
+    exit
+  fi
+
+  report_main_step "Differences found. Checking for an existing pull request."
+
+  KIBANA_MACHINE_USERNAME="kibanamachine"
+  git config --global user.name "$KIBANA_MACHINE_USERNAME"
+  git config --global user.email '42973632+kibanamachine@users.noreply.github.com'
+
+  PR_TITLE='[ES|QL] Update grammars'
+  PR_BODY='This PR updates the ES|QL grammars (lexer and parser) to match the latest version in Elasticsearch.'
+
+  # Check if a PR already exists
+  pr_search_result=$(gh pr list --search "$PR_TITLE" --state open --author "$KIBANA_MACHINE_USERNAME"  --limit 1 --json title -q ".[].title")
+
+  if [ "$pr_search_result" == "$PR_TITLE" ]; then
+    echo "PR already exists. Exiting."
+    exit
+  fi
+
+  echo "No existing PR found. Proceeding."
+
+  report_main_step "Building ANTLR artifacts."
+
+  # Bootstrap Kibana
+  yarn kbn bootstrap 
+
+  # Build ANTLR stuff
+  cd ./packages/kbn-monaco/src  
+  yarn build:antlr4:esql 
+
+  # Make a commit
+  BRANCH_NAME="esql_grammar_sync_$(date +%s)"
+
+  git checkout -b "$BRANCH_NAME"
+
+  git add -A
+  git commit -m "Update ES|QL grammars"
+
+  report_main_step "Changes committed. Creating pull request."
+
+  git push --set-upstream origin "$BRANCH_NAME"
```

Comment:
To keep origin to a minimal set of branches:

```suggestion
  git remote add kibanamachine https://github.com/kibanamachine/kibana.git
  git push kibanamachine "$BRANCH_NAME"
```

Reply:
Done in https://github.com/elastic/kibana/pull/178347/commits/b57b7e5615e3c4701e8e4e81199f8b43681dc710